src/D3MHub.sol

```diff
@@ -221,121 +219,121 @@ contract D3MHub {
+            daiJoin.exit(address(_pool), amount);
+            _pool.deposit(amount);
         }
-
-        vat.slip(ilk, address(_pool), int256(amount));
-        vat.frob(ilk, address(_pool), address(_pool), address(this), int256(amount), int256(amount));
-        // normalized debt == erc20 DAI (Vat rate for D3M ilks fixed to 1 RAY)
-        daiJoin.exit(address(_pool), amount);
-        _pool.deposit(amount);
-
         emit Wind(ilk, amount);
     }
 
-    function _unwind(
-                bytes32 ilk,
-                ID3MPool _pool,
-                uint256 supplyReduction, // [wad]
-                Mode mode,
-                uint256 assetBalance     // [wad]
-             ) internal {
-        // IMPORTANT: this function assumes Vat rate of D3M ilks will always be == 1 * RAY (no fees).
-        // That's why it converts normalized debt (art) to Vat DAI generated with a simple RAY multiplication or division
-        // This module will have an unintended behaviour if rate is changed to some other value.
-
-        EndLike _end;
-        uint256 daiDebt;
-        if (mode == Mode.NORMAL) {
-            // Normal mode or module just caged (no culled)
-            // debt is obtained from CDP art
-            (, daiDebt) = vat.urns(ilk, address(_pool));
-        } else if (mode == Mode.D3M_CULLED) {
-            // Module shutdown and culled
-            // debt is obtained from free collateral owned by the pool contract
-            // We rebalance the CDP after grabbing in `cull` so the gems represents
-            // the debt at time of cull
-            daiDebt = vat.gem(ilk, address(_pool));
-        } else if (mode == Mode.MCD_CAGED) {
-            // MCD caged
-            // debt is obtained from free collateral owned by the End module
-            _end = end;
-            _end.skim(ilk, address(_pool));
-            daiDebt = vat.gem(ilk, address(_end));
-        } else revert("D3MHub/unknown-mode");
-
-        uint256 availableAssets = _pool.maxWithdraw();
+    function _unwind(bytes32 ilk, ID3MPool _pool, uint256 amount) internal {
+        if (amount > 0) {
+            _pool.withdraw(amount);
+            daiJoin.join(address(this), amount);
+            vat.frob(ilk, address(_pool), address(_pool), address(this), -int256(amount), -int256(amount));
+            vat.slip(ilk, address(_pool), -int256(amount));
+        }
+        emit Unwind(ilk, amount);
+    }
 
-        // Unwind amount is limited by how much:
-        // - max reduction desired
-        // - assets available
-        // - dai debt tracked in vat (CDP or free)
-        uint256 amount = _min(
-                            _min(
-                                supplyReduction,
-                                availableAssets
-                            ),
-                            daiDebt
-                        );
-        require(amount <= MAXINT256, "D3MHub/overflow");
+    function _fullUnwind(bytes32 ilk, ID3MPool _pool, address urn) internal {
```

Comment:
Daiwanese attempt: `unfurl`
```suggestion
    function _unfurl(bytes32 ilk, ID3MPool _pool, address urn) internal {
```

Reply:
This is totally forced and not even daiwanese. I proceeded to just overload the method `unwind`.

Reply:
yeah, the attempt was more to reduce name size with a synonym of `unwind`, overloading the method sounds better.